src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/SingleSpaceSeparatorCheck.java

```diff
@@ -127,7 +127,9 @@ public boolean isCommentNodesRequired() {
 
     @Override
     public void beginTree(DetailAST rootAST) {
-        visitEachToken(rootAST);
+        if (rootAST != null) {
```

Comment:
Changing other method for mutation required this to be changed as well when regression was run.